compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -30,6 +30,11 @@ import dotty.tools.io.{AbstractFile, JarArchive}
 import dotty.tools.dotc.util.Property
 import dotty.tools.dotc.semanticdb.DiagnosticOps.*
 import scala.util.{Using, Failure, Success}
+import com.google.protobuf.Empty
+import com.google.protobuf.UnknownFieldSet
+import com.google.protobuf.UnknownFieldSet.Field
```

Comment:
these are java protobuf classes - you should be able to get a slightly better syntax (but no significant difference in performance) with classes in the `scalapb` package

Reply:
Unfortunately, we can't use scalapab runtime, since we intentially removes the dependency to that package in `semanticdb-for-scala3` since we don't want scala3 compiler to include scalapb runtime and it's transitive dependencies. (TBH, I have no idea why the compiler has an `com.google.protobuf` dependency 🤔 )

Reply:
it doesnt work anyway,

If we want to do this then we should copy these definitions to the `dotty.tools.dotc.semanticdb.internal` package
<img width="1115" alt="Screenshot 2023-07-24 at 15 05 19" src="https://github.com/lampepfl/dotty/assets/13436592/cfb74fd5-7cef-4770-a397-28888f80835b">

Reply:
there used to be a dependency on protobuf due to Zinc 1.3, but we merged yesterday depending exclusively on Zinc 1.9 which no longer has protobuf

Reply:
> it doesnt work anyway,

`com.google.protobuf` removed recently? It works in my local environment 🤔 

> there used to be a dependency on protobuf due to Zinc 1.3, but we merged yesterday 

Ah, ok.

Reply:
I'm on the fence wether should we do without `com.google.protobuf`.
Should we still being performant by reading protobuf without parsing?

I gonna look into is there an API that reads protobuf as an Empty in `scalapb` that doesn't depend on `com.google.protobuf`, but I'm going to hold off from this PR for a while since I don't have enough band-width for this 😓

Reply:
Could we merge the current state or one with the fixes? This would unblock scalafix, so would be pretty great to do. We could do optimization improvements later?

Reply:
I'll move back to "parsing TextDocuments and update it" anyway :)